src/webgpu/api/validation/queue/copyToTexture/CopyExternalImageToTexture.spec.ts

```diff
@@ -184,15 +374,80 @@ g.test('source_imageBitmap,state')
     if (closed) imageBitmap.close();
 
     t.runTest(
-      { imageBitmap },
+      { source: imageBitmap },
       { texture: dstTexture },
       copySize,
       true, // No validation errors.
       closed ? 'InvalidStateError' : ''
     );
   });
 
+g.test('source_offscreenCanvas,state')
+  .desc(
+    `
+  Test OffscreenCanvas as source image in state [valid, detached].
+
+  Transfer OffsreenCanvas to worker will detach the OffscreenCanvas
+  in main thread.
+  
+  Check whether 'InvalidStateError' is generated when OffscreenCanvas is
+  detached.
+  `
+  )
+  .params(u =>
+    u //
+      .combine('detached', [false, true])
+      .beginSubcases()
+      .combine('copySize', [
+        { width: 0, height: 0, depthOrArrayLayers: 0 },
+        { width: 1, height: 1, depthOrArrayLayers: 1 },
+      ])
+  )
+  .fn(async t => {
+    const { detached, copySize } = t.params;
+    const offscreenCanvas = t.getOffscreenCanvas(1, 1);
+    const dstTexture = t.device.createTexture({
+      size: { width: 1, height: 1, depthOrArrayLayers: 1 },
+      format: 'bgra8unorm',
+      usage: GPUTextureUsage.COPY_DST | GPUTextureUsage.RENDER_ATTACHMENT,
+    });
+
+    if (detached) {
+      const workerCode =
```

Comment:
You don't actually need to create a worker in order to post to it, you can just create a MessageChannel and post to it:

```ts
const messageChannel = new MessageChannel();
messageChannel.port1.postMessage(offscreenCanvas, [offscreenCanvas]);
```

Reply:
One remain issue here. I try to construct a test case that port1 listening to the port2 message which do context construction for the offscreenCanvas. But I failed to run it due to the case will terminate(return all of the webgpu resources) before the onmessage callback has been called which runs the `runTest` function.
I tried several ways but I don't think current cts support test case waits for a callback and then terminate.